.github/workflows/rust.yml

```diff
@@ -0,0 +1,54 @@
+on: [push, pull_request]
+
+name: Continuous integration
+
+jobs:
+    tests:
+        name: Tests
+        runs-on: ubuntu-latest
+        strategy:
+            matrix:
+                include:
+                    - rust: stable
+                      env:
+                          RUSTFMTCHK: true
+                    - rust: nightly
+                      env:
+                          RUSTFMTCHK: false
+                    - rust: 1.29.0
+                      env:
+                          PIN_VERSIONS: true
+        steps:
+            - name: Checkout Crate
+              uses: actions/checkout@v2
+            - name: Checkout Toolchain
+              uses: actions-rs/toolchain@v1
+              with:
+                  profile: minimal
+                  toolchain: ${{ matrix.rust }}
+                  override: true
+            - name: Running test script
+              env: ${{ matrix.env }}
+              run: ./contrib/test.sh
+
+    integrations-tests:
+        name: Integration Tests
+        runs-on: ubuntu-latest
+        strategy:
+            matrix:
+                rust: [stable]
+                bitcoinversion:
+                    [0.18.0, 0.18.1, 0.19.0.1, 0.19.1, 0.20.0, 0.20.1, 0.21.0]
```

Comment:
I recommend putting these in quotes. Even though this works fine for these versions once we add 22.0 it will try to download "22" instead of "22.0".

Reply:
Good point fixed